src/figuresClasses.ts

```diff
@@ -1,11 +1,92 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
You don't need to manually assign constructor parameters to class properties if you're using TypeScript's parameter properties. Simply declare the parameters with access modifiers, and TypeScript will handle the property creation.